src/interface/web/app/share/chat/page.tsx

```diff
@@ -276,11 +239,66 @@ export default function SharedChat() {
                     timestamp: (new Date()).toISOString(),
                     rawQuery: queryToProcess || "",
                 }
+                setProcessQuerySignal(true);
                 setMessages(prevMessages => [...prevMessages, newStreamMessage]);
             }
         })();
     }, [conversationId]);
 
+    function processMessageChunk(rawChunk: string) {
```

Comment:
`processMessageChunk` and some other related functions seem to be defined twice, here on the `share/page.tsx` and `chat/page.tsx` Can they be deduplicated into `common/chatFunctions.ts` to ease maintenance?

Reply:
It's a little bit tricky because they're also doing a bunch of state manipulation, but I'll see what can be improved here.